### PR TITLE
roachtest: move utilities to roachtestutil

### PR DESCRIPTION
**Pull request by andrewbaptist**
Previously many of the test utilities were inside the tests package. This forced any tests to also be inside that package. By moving them into roachtestutil (and making public) it becomes easier to split tests out.

Epic: None

Release note: None

### CODE REVIEW

**Comment by andrewbaptist** [comment]
This change allows moving roachtests to separate directories as they no longer rely on private functions inside the `tests` directory. It doesn't have any functional changes.

File: pkg/cmd/roachtest/roachtestutil/util.go
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/util.go:1]: There is an existing `utils.go` in the same package; is there a reason not to merge these helpers into it (or vice versa)?

File: pkg/cmd/roachtest/roachtestutil/util_if_local.go [hunk 1]
```diff
@@ -3,11 +3,11 @@
 // Use of this software is governed by the CockroachDB Software License
 // included in the /LICENSE file.
 
-package tests
+package roachtestutil
 
 import "github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
 
-func ifLocal(c cluster.Cluster, trueVal, falseVal string) string {
+func IfLocal(c cluster.Cluster, trueVal, falseVal string) string {
```
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/util_if_local.go:10]: Does it need to be in a separate file?

File: pkg/cmd/roachtest/roachtestutil/util_load_group.go
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/util_load_group.go:1]: Everything is already a util given the new package. Should we maybe drop the `util_` prefix from the source names?

**Review by srosenberg — APPROVED** [review]
Looks good modulo a few nits. I've kicked off [SELECT_PROBABILITY=0.25](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17631419?buildTab=log&logFilter=debug&focusLine=12&logView=linear) as a sanity check.

**Review by andrewbaptist — COMMENTED** [review]
TFTR!

I made changes and will bors it once CI goes through.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134209#-:-OB6REO23Euo9HzraqpF:b-hdmcrv)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @herkolategan, and @srosenberg)

___
*[`pkg/cmd/roachtest/roachtestutil/util.go` line 0 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134209#gh-1831922873:-OB6T4oX-2dC7h_J0-Af:b-zae3aq) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb65884531ba3835efb67f66a64de60c9709865f/pkg/cmd/roachtest/roachtestutil/util.go#L0)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

There is an existing `utils.go` in the same package; is there a reason not to merge these helpers into it (or vice versa)?
</blockquote></details>

I removed the util.go and created a profile.go which includes the pieces specific to profiling. I moved the rest into the utils pkg. 
___
*[`pkg/cmd/roachtest/roachtestutil/util_if_local.go` line 10 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134209#gh-1831923557:-OB6TDVMCVT075c4oAr4:brnxklp) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb65884531ba3835efb67f66a64de60c9709865f/pkg/cmd/roachtest/roachtestutil/util_if_local.go#L10)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Does it need to be in a separate file?
</blockquote></details>

Agreed! . I moved this function into utils.go
___
*[`pkg/cmd/roachtest/roachtestutil/util_load_group.go` line 0 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134209#gh-1831924199:-OB6TLbyAqOLLmRfyspY:bd59ic7) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb65884531ba3835efb67f66a64de60c9709865f/pkg/cmd/roachtest/roachtestutil/util_load_group.go#L0)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Everything is already a util given the new package. Should we maybe drop the `util_` prefix from the source names?
</blockquote></details>

Agreed! I removed the util_ prefix.

**Comment by andrewbaptist** [comment]
bors r=srosenberg

**Pull request merged**